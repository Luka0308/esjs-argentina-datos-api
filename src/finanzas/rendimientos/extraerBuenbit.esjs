importar { load } desde 'cheerio'
importar { format } desde 'date-fns'
importar axios desde 'axios'

exportar asincrono funcion extraerBuenbit() {
  intentar {
    const respuesta = esperar axios.get(
      import.meta.env.VITE_FINANZAS_RENDIMIENTOS_BUENBIT_URL,
    )

    si (respuesta.data.errors.longitud > 0) {
      retornar []
    }

    const items = []

    const fecha = format(crear Fecha(), 'yyyy-MM-dd')

    para (const item de respuesta.data.object) {
      const moneda = item.currency.aMayusculas()
      const apy = Numero(item.apy)

      items.agregar({
        moneda,
        apy,
        fecha,
      })
    }

    retornar items
  } capturar (error) {
    consola.error(error)
    retornar []
  }
}
