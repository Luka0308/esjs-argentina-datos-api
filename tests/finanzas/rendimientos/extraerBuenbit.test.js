import { describe, expect, it } from 'vitest'
import { extraerBuenbit } from '@/finanzas/rendimientos/extraerBuenbit.esjs'
// import { guardarRendimientos } from '@/finanzas/rendimientos/guardarRendimientos.esjs'

describe('extraerBuenbit', () => {
  it('extrae los rendimientos', async () => {
    try {
      const items = await extraerBuenbit()

      expect(items).toMatchSnapshot()
    } catch (error) {
      console.error(error)
    }
  })
})
