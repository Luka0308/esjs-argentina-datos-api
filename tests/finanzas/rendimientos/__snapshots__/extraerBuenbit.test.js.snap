// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`extraerBuenbit > extrae los rendimientos 1`] = `
[
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "ARS",
  },
  {
    "apy": 3.5,
    "fecha": "2024-02-15",
    "moneda": "DAI",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "USD",
  },
  {
    "apy": 1.75,
    "fecha": "2024-02-15",
    "moneda": "ETH",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "BTC",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "PEN",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "USD",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "ADA",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "BNB",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "MATIC",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "DOT",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "SOL",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "MXN",
  },
  {
    "apy": 122.36,
    "fecha": "2024-02-15",
    "moneda": "NUARS",
  },
  {
    "apy": 4,
    "fecha": "2024-02-15",
    "moneda": "USDT",
  },
  {
    "apy": 6.5,
    "fecha": "2024-02-15",
    "moneda": "USDC",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "BUSD",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "UST",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "LUNA",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "PAXG",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "MANA",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "AXS",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "FTM",
  },
  {
    "apy": 4,
    "fecha": "2024-02-15",
    "moneda": "AVAX",
  },
  {
    "apy": 4,
    "fecha": "2024-02-15",
    "moneda": "NUPEN",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "RIF",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "RBTC",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "RDOC",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "COP",
  },
  {
    "apy": 7,
    "fecha": "2024-02-15",
    "moneda": "NCOP",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "LAC",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "UXD",
  },
  {
    "apy": 3.5,
    "fecha": "2024-02-15",
    "moneda": "USDM",
  },
  {
    "apy": 0,
    "fecha": "2024-02-15",
    "moneda": "USD",
  },
]
`;
